packages/odata-common/src/entity-deserializer.ts

```diff
@@ -79,7 +79,7 @@ export function entityDeserializer<T extends DeSerializers>(
     requestHeader?: any
   ): EntityT {
     const etag = extractODataETag(json) || extractEtagFromHeader(requestHeader);
-    return Object.values(entityApi.schema)
+    return Object.values(entityApi.schema as Record<string, any>)
```

Comment:
here you enforce the type you removed in entityBase - can it stay in entityBase?

Reply:
The idea was to improve the types on `testEntityApi.schema.TO_SINGLE_LINK._linkedEntityApi.schema...` but it didn't work ;) This should be fixed on your PR.